flink-dist/src/main/assemblies/hadoop.xml

```diff
@@ -28,13 +28,14 @@ under the License.
 	<includeBaseDirectory>true</includeBaseDirectory>
 	<baseDirectory>flink-${project.version}</baseDirectory>
 
-	<files>
-		<!-- copy the Hadoop uber jar -->
-		<file>
-			<source>../flink-shaded-hadoop/flink-shaded-hadoop2-uber/target/flink-shaded-hadoop2-uber-${hadoop.version}-${project.version}.jar</source>
+	<dependencySets>
+		<dependencySet>
 			<outputDirectory>lib/</outputDirectory>
-			<destName>flink-shaded-hadoop2-uber-${project.version}-${hadoop.version}.jar</destName>
-			<fileMode>0644</fileMode>
-		</file>
-	</files>
+			<useTransitiveDependencies>true</useTransitiveDependencies>
+			<scope>provided</scope>
```

Comment:
we could add `<outputFileNameMapping>${artifact.artifactId}-${artifact.baseVersion}.${artifact.extension}</outputFileNameMapping>
`, so that (for SNAPSHOT builds) the jar is always called `flink-shaded-hadoop2-uber-2.4.1-1.9-SNAPSHOT.jar`, and not `flink-shaded-hadoop2-uber-2.4.1-1.9-629462874621478328407-23.jar`

Reply:
I think If we add the dependency for `flink-shaded-hadoop-2-uber` with `<version>${hadoop.version}-${flink.shaded.version}</version>`. we never using the SANAPSHOT JRA. e.g.: `flink-shaded-hadoop-2-uber-2.4.1-7.0.jar`Is there anything I misunderstand?